### ISSUE TITLE
Non-idempotent formatting with custom align.tokens

### ISSUE THREAD

**Issue by acjay** [scalameta/scalafmt#1220]
This template is a guideline, not a strict requirement.
- **Version**: 1.15
- **Integration**: SBT
- **Configuration**:
```
align = more
align.openParenCallSite = false
align.openParenDefnSite = false
align.tokens.add = [{code = ":", owner = "Term.Param"}, "=", "shouldBe"]
continuationIndent.callSite = 2
continuationIndent.defnSite = 2
danglingParentheses = true
docstrings = JavaDoc
importSelectors = binpack
maxColumn = 80
project.excludeFilters = ["net/artsy/auction/db", "protobuf"]
rewrite.rules = [SortImports]
spaces.inImportCurlyBraces = true
style = defaultWithAlign
unindentTopLevelOperators = true
```

## Steps

Given code like this:
```scala
case class AggregatedInfo(
    currentLotId: Option[LotId],
    fullSaleState: FullSaleState,
    saleStatistics: Option[SaleStatistics],
    userInfo: UserInfo,
    saleOnHoldMessage: Option[String]
)
```

When I run scalafmt like this:

```bash
scalafmt --config .scalafmt.conf .
```

or via SBT

## Problem

Scalafmt oscilaltes between that formatting and the desired formatting of.

## Expectation

I would like the formatted output to look like this:

```scala
case class AggregatedInfo(
  currentLotId:      Option[LotId],
  fullSaleState:     FullSaleState,
  saleStatistics:    Option[SaleStatistics],
  userInfo:          UserInfo,
  saleOnHoldMessage: Option[String]
)
```

## Workaround

I've found that by running scalafmt repeatedly, it sometimes converges to what I want, but upon next compile, it often reverts back. In other words, it appears to be oscillating between states, which is producing noisy diffs.

## Notes

Sensible columnar alignment is a major goal of our Scalafmt usage. Scalariform does this in some cases, like parameter declarations, and it is encouraging the Scalafmt in principle could be more powerful. But in practice, the oscillatory behavior makes this really problematic. It's worse than needing to [run mutliple times](https://github.com/scalameta/scalafmt/issues/1055) because it never quite converges.

**Comment by olafurpg** [comment]
Thanks for reporting! This is a familiar issue and is unlikely to change unless you are motivated to dig into the alignment code yourself. Until then, I recommend using the builtin `align.tokens` values.

**Comment by sherifkandeel** [comment]
on a separate note, where could one find the possible values of `owner` for each `token`? 

I would like to do something similar but only within params of one class, not across multiple classes.

so not: 
```
case class One(param1:       String)
case class Two(longerParam1: String)
```
but: 
```
case class AggregatedInfo(
  currentLotId:      Option[LotId],
  fullSaleState:     FullSaleState,
  saleStatistics:    Option[SaleStatistics],
  userInfo:          UserInfo,
  saleOnHoldMessage: Option[String]
)
```

**Comment by olafurpg** [comment]
@sherifkandeel that is another known limitation of the alignment engine and one reason why `Term.Param` is not enabled as part of `align = most`.

**Comment by poslegm** [comment]
I can't reproduce it with v2.4.2. Is issue still actual?

**Comment by falconepl** [comment]
@poslegm It might be fixed already, as #1309 has been closed about a month ago
